app/extensions/__init__.py

```diff
@@ -92,6 +95,54 @@ def process_result_value(self, value, dialect):
             return json.loads(value)
 
 
+SA_JSON = db.JSON
+
+
+def custom_json_decoder(obj):
+    for key, value in obj.items():
+        if isinstance(value, str) and re.match(
+            '^[A-Z][a-z][a-z], [0-9][0-9] [A-Z][a-z][a-z]', value
+        ):
+            try:
+                obj[key] = datetime.strptime(value, '%a, %d %b %Y %H:%M:%S %Z')
+            except ValueError:
+                pass
+    return obj
+
+
+class JSON(db.TypeDecorator):
+    impl = SA_JSON
+
+    def process_bind_param(self, value, dialect):
+        # Adapted from sqlalchemy/sql/sqltypes.py JSON.bind_processor
```

Comment:
Can you explain what the adaptation was that was needed rather than using the base one, was it simply the object hook for the deserialiser (line 136)

Reply:
For `process_bind_param`, it's using `flask.json.JSONEncoder` instead of `json.dumps`

Reply:
For `process_result_value`, it is the `custom_json_encoder`